### PR TITLE
Change updraft momentum equation to affective form

### PR DESCRIPTION
**Pull request by yairchn**
After #914 I am starting a new PR for advective updraft vertical velocity. I have removed `w` from the auxiliary variables in the updrafts and made the corresponding changes. Most notable changes in the PR are the change of the updraft `w` equation in `compute_updraft_tendencies` and the corresponding changes to the non hydrostatic pressure term, which entails some change is the non hydrostatic pressure tendency in the TKE equation (in `update_aux.jl`).

### CODE REVIEW

File: src/TurbulenceConvection/EDMF_functions.jl [hunk 1]
```diff
@@ -780,12 +781,12 @@ function compute_up_tendencies!(
...
-        @. tends_ρaw = -(∇f(wvec(LBC(ρaw * w_up))))
-        @. tends_ρaw +=
-            (ρaw * (I0f(entr_w) * w_en - I0f(detr_w) * w_up)) +
-            (ρ_f * Iaf(a_up) * I0f(buoy)) +
-            nh_pressure
-        tends_ρaw[kf_surf] = 0
+        @. tends_w = -(∇f(wvec(LBC(w * w_up))))
+        @. tends_w +=
+            w * I0f(entr_w) * (w_en - w_up) +
+            I0f(buoy) +
```
Comment (yairchn) [inline src/TurbulenceConvection/EDMF_functions.jl:787]: here I am dividing by `(ρ_f * Iaf(max(a_up, a_min)))` and multiplying again in `nh_pressure`. This should be removed. The main challenge I see here is that in TRMM I `w>0` for `area` slightly above the minimum area for which we filter. The `w` feedbacks through  `nh_pressure` and grows to unphysical values.

**Comment by trontrytel** [comment]
@yairchn I see you already increased the reference counter in regression tests. Do you remember which cases were falgged as different?

**Comment by trontrytel** [comment]
> After #914 I am starting a new PR for advective updraft w.

Does it mean https://github.com/CliMA/ClimaAtmos.jl/pull/902 should be closed?

**Comment by trontrytel** [comment]
> @yairchn I see you already increased the reference counter in regression tests. Do you remember which cases were falgged as different?

The biggest difference I can spot is in the buoyancy for the convective cases (Bomex and TRMM):

Before / after compressible Bomex

Screenshot 2022-10-09 at 16 22 04

Screenshot 2022-10-09 at 16 15 59

Before / after: anelastic TRMM:

Screenshot 2022-10-09 at 16 21 15

Screenshot 2022-10-09 at 16 17 17

**Comment by yairchn** [comment]
> @yairchn I see you already increased the reference counter in regression tests. Do you remember which cases were falgged as different?

opps, I shouldn't have done it yet. can I undo it ? @charleskawczynski ?

**Comment by yairchn** [comment]
> > After #914 I am starting a new PR for advective updraft w.
> 
> Does it mean #902 should be closed?

yes!

File: src/TurbulenceConvection/EDMF_functions.jl [hunk 2]
```diff
@@ -368,7 +368,7 @@ function set_edmf_surface_bc(
             prog_up[i].ρaq_liq[kc_surf] = prog_up[i].ρarea[kc_surf] * q_liq_surf
             prog_up[i].ρaq_ice[kc_surf] = prog_up[i].ρarea[kc_surf] * q_ice_surf
         end
-        prog_up_f[i].ρaw[kf_surf] = ρ_f[kf_surf] * w_surface_bc(surf)
```
Comment (trontrytel) [inline src/TurbulenceConvection/EDMF_functions.jl:371]: Were we missing area fraction here before?

Comment (yairchn) [reply src/TurbulenceConvection/EDMF_functions.jl:371]: > Were we missing area fraction here before?

yes but `w_surface_bc(surf) = 0` we are lucky ;-)

**Comment by charleskawczynski** [comment]
> opps, I shouldn't have done it yet. can I undo it ? @charleskawczynski ?

Yes, I'm fine with doing whatever is needed!

File: src/TurbulenceConvection/EDMF_functions.jl [hunk 3]
```diff
@@ -780,12 +781,12 @@ function compute_up_tendencies!(
...
 
-        @. tends_ρaw = -(∇f(wvec(LBC(ρaw * w_up))))
-        @. tends_ρaw +=
-            (ρaw * (I0f(entr_w) * w_en - I0f(detr_w) * w_up)) +
-            (ρ_f * Iaf(a_up) * I0f(buoy)) +
-            nh_pressure
-        tends_ρaw[kf_surf] = 0
+        @. tends_w = -(∇f(wvec(LBC(w * w_up))))
+        @. tends_w +=
+            w * I0f(entr_w) * (w_en - w_up) +
```
Comment (trontrytel) [inline src/TurbulenceConvection/EDMF_functions.jl:786]: We used to have `detr_w` multiplying `w_up` here. Now there is only `entr_w`. Is that on purpose?

**Comment by yairchn** [comment]
@trontrytel thanks for the comparison - I haven't got to compare that yet, I am trying to resolve the issue I commented o n with deciding and multiplying by area fraction of the pressure term.

File: src/TurbulenceConvection/EDMF_functions.jl (hunk 2 above)
Comment (trontrytel) [reply src/TurbulenceConvection/EDMF_functions.jl:371]: phew...

File: src/TurbulenceConvection/EDMF_functions.jl (hunk 3 above)
Comment (yairchn) [reply src/TurbulenceConvection/EDMF_functions.jl:779]: yes, see equation (21) in Tan et al
Screen Shot 2022-10-09 at 6 34 39 PM

**Review by trontrytel — CHANGES_REQUESTED** [review]
The only question I have is about the `detr_w` potentially missing around line 777 in `EDMF_functions.jl`. I think the changes to the buoyancy profiles are acceptable. We should just remember to also look at TRMM long run (compressible simulation) when it is done

File: src/TurbulenceConvection/EDMF_functions.jl (hunk 3 above)
Comment (trontrytel) [reply src/TurbulenceConvection/EDMF_functions.jl:779]: So the detrainment was only there through density changes?

**Comment by trontrytel** [comment]
> @trontrytel thanks for the comparison - I haven't got to compare that yet, I am trying to resolve the issue I commented o n with deciding and multiplying by area fraction of the pressure term.

Sounds good. I'll take another look at the plots when you are done with the pressure then. But for the (limited) number of single column cases we have here nothing bad really stands out. So I think this looks great

**Review by trontrytel — APPROVED** [review]

File: src/TurbulenceConvection/EDMF_functions.jl (hunk 1 above)
Comment (yairchn) [reply src/TurbulenceConvection/EDMF_functions.jl]: I forgot that `nh_pressure` is also used in the TKE equation. Once this is done consistently I think the PR is ready

File: src/TurbulenceConvection/update_aux.jl [hunk 4]
```diff
@@ -556,7 +556,7 @@ function update_aux!(
     @inbounds for i in 1:N_up
         w_up = aux_up_f[i].w
         nh_press = aux_up_f[i].nh_pressure
-        @. tke_press += (Ic(w_en) - Ic(w_up)) * Ic(nh_press)
+        @. tke_press += (Ic(w_en) - Ic(w_up)) * prog_up[i].ρarea * Ic(nh_press)
```
Comment (trontrytel) [inline src/TurbulenceConvection/update_aux.jl:559]: Why are we multiplying by `prog_up[i].ρarea` here now?

**Review by charleskawczynski — APPROVED** [review]

**Comment by yairchn** [comment]
Bors r+

**Pull request merged**